### DISCUSSION TITLE
Combobox: Option to render html element inside selected item

### DISCUSSION THREAD

**Discussion by kalaindrek** [twilio-labs/paste#953] [Ideas] [answered]
Hello team

The combobox supports `optionTemplate` to display more complex items in the list box, but when the combobox is closed then only string representation can be shown.

How it looks like today
<img width="575" alt="1" src="https://user-images.githubusercontent.com/24303628/99508080-dde71000-298c-11eb-8725-1f14b9e76e0f.png">

Expected
<img width="575" alt="2" src="https://user-images.githubusercontent.com/24303628/99508109-e63f4b00-298c-11eb-863d-25d7ac731c3c.png">

Here is the code sample: https://codesandbox.io/s/combobox-hqy2v
Is there plans to support this functionality?

**Comment by SiTaggart** [comment, marked as answer]
Hi @kalaindrek 

There is a constraint that we are adhering to with the Combobox, and that is it's an HTML input element that stores the value. As such we are limited to what an HTML input element can display as a value, it can only display strings.

You can get the first part fairly easily by updating the `itemToString` method you supply

```
itemToString={(item: any) => (item ? `${String(item.label)} | String(item.value)` : null)}
```

Where you concat the item and value together into a single string to display. 

We just can't support displaying an icon (or anything other than a string) as part of the displayed value.

Hope that answers your question.
